### DISCUSSION TITLE
Is there a way to improve the performance of `git log --shortstat`

### DISCUSSION THREAD

**Discussion by eamodio** [git-for-windows/git#4418] [Q&A]
In GitLens (and in GitKraken Client with the new experimental feature to use the Git binary) we run:

`git log --stdin -z --format=%x00%x00%H%x00%aN%x00%aE%x00%at%x00%ct%x00%P%x00%D%x00%B%x00 --date-order --all --`

Which is nicely optimized when the `commit-graph` Git feature is enabled, but when you add `--shortstat` to that it becomes 10x slower. Is there any feature in Git to optimize that?

For reference we use `--shortstat` to provide the visualization data for the minimap show above the Commit Graph, and for the `Changes` column so that you can see the magnitude of the change set.

![image](https://github.com/git-for-windows/git/assets/641685/afc3898b-7ac7-4f70-bd7e-af4e921786d2)

**Comment by dscho** [comment]
I am not aware of such a feature. But then, this is not a Windows-specific question. Maybe take it to [the Git mailing list](mailto:git@vger.kernel.org) (send plain-text messages, HTML messages are dropped silently)?

**Comment by ffyuanda** [comment]
@eamodio Hi, I ran a bunch of `perf` tests and Git's `trace2` API, and found that because every commit needs to be diff'd to show the stats (whether it's `shortstat` or any stats), I don't _think_ there is a way to overcome this overhead brought by diffing commits. 

The extra cost of `shortstat` vs without `shortstat` seems to be exactly caused by diffing and its related overhead. This cost is roughly O(n), where n is the number of commits you request.

Please correct me if I'm wrong.

You can check these functions for more info:
[log_tree_diff](https://github.com/git/git/blob/0df2c180904f6b709766f9c24669a9d01543f915/log-tree.c#L1036)
[diffcore_std](https://github.com/git/git/blob/0df2c180904f6b709766f9c24669a9d01543f915/diff.c#L6890)
